### PR TITLE
Automatic Mobile Resumption Hash Management

### PR DESCRIPTION
**Pull request by joeljfischer**
Add automatic hash management to the manager layer of the libraries. Automatically send the resumption hash upon connection and refill the managers with the data they contained at the time that hash was retrieved.

### CODE REVIEW

**Comment by joeljfischer** [comment]
@theresalech this is ready for review

File: proposals/nnnn-mobile-hash-managment.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+* Status: **Awaiting review**
+* Impacted Platforms: iOS / Android
+
+## Introduction
+
+Add automatic hash management to the manager layer of the libraries. Automatically send the resumption hash upon connection and refill the managers with the data they contained at the time that hash was retrieved.
+
+## Motivation
+
+With the addition of the manager layer, hashes have become significantly more difficult to use. While hashes can be sent and the data will resume on the system, the manager-layer does not know this and cannot restore the data that is currently in use. This may cause bugs or for data to be resent wholesale that merely needs to be updated.
```
Comment (theresalech) [inline proposals/nnnn-mobile-hash-managment.md]: Can you please reword to "...to be resent in its entirety that merely..."

File: proposals/nnnn-mobile-hash-managment.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+// The choice set is already handled by a delegate
+
+@end
+```
+
+Resumed data would **only** fire through this delegate.
+
+### Android
+
+Android will have to make similar changes and updates to the (yet to come) manager layer. Android has SQLite built in, but will likely store a JSON object into SharedPreferences when the managers disconnect and loaded on connection. A similar delegate-style interface may have to be added to handle restored commands.
```
Comment (theresalech) [inline proposals/nnnn-mobile-hash-managment.md]: Should this say "...when the managers disconnect and load on connection." ?

**Comment by joeljfischer** [comment]
@theresalech All fixed!

**Pull request merged**